openmc/mesh.py

```diff
@@ -229,14 +229,17 @@ def write_data_to_vtk(self, points, filename, datasets, volume_normalization=Tru
         from vtk.util import numpy_support as nps
 
         # check that the data sets are appropriately sized
-        errmsg = "The size of the dataset {} should be equal to the number of cells"
         for label, dataset in datasets.items():
+            errmsg = (
+                f"The size of the dataset '{label}' ({dataset.size}) should be"
+                f" equal to the number of mesh cells ({self.num_mesh_cells})"
+            )
             if isinstance(dataset, np.ndarray):
-                if not dataset.size == self.dimension[0] * self.dimension[1]* self.dimension[2]:
-                    raise RuntimeError(errmsg.format(label))
+                if not dataset.size == self.num_mesh_cells:
+                    raise RuntimeError(errmsg)
```

Comment:
I would recommend using [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError) for this exception, since it's basically saying that the _value_ of an argument is not right.

Reply:
Certainly